ffi/mupdf.lua

```diff
@@ -664,7 +664,7 @@ function page_mt.__index:draw_new(draw_context, width, height, offset_x, offset_
 end
 
 mupdf.STRIKE_HEIGHT = 0.375
-mupdf.UNDERLINE_HEIGHT = 0.075
+mupdf.UNDERLINE_HEIGHT = 0.01
```

Comment:
That's quite a difference :|
What does this value mean exactly, if you figured it out ?
But as long as it looks OK, I don't mind not knowing :)

Reply:
It moves the line up and down

Reply:
0.2
![10](https://user-images.githubusercontent.com/62179190/151033036-48692861-0c16-4d9d-9a62-96d3128d1555.png)

Reply:
> 0.2

What is the black and what the grey line ?

What is the reference ? Where it is drawn with mupdf.UNDERLINE_HEIGHT = 0 ?

Reply:
0.05
![11](https://user-images.githubusercontent.com/62179190/151033628-83fcb314-46f0-4287-9278-444384c6dc42.png)

Reply:
> What is the black and what the grey line ?

Black - mupdf, grey - our highlight.

Reply:
Sorry, still unclear: is that MuPDF only or both KOReader + MuPDF drawings ?
Why it feels there is a black line over a gray line with 0.01 and 0.05 ? KOReader dimming some part of MuPDF highlight making it blacker ? Or MuPDF doing it black, but KOReader drawing some gray over it or part of it ?

I'm unsure if 0 feels like it's the bottom of the line box ?
mupdf.UNDERLINE_HEIGHT is really a delta y and not a line height ?

(But 0 looks better no ? only gray, vs bits of black with 0.01 and 0.05).

And just curious: what would be drawn with mupdf.UNDERLINE_HEIGHT = 1 ?

Reply:
> MuPDF doing it black, but KOReader drawing some gray over it or part of it ?

MuPDF draws saved (in the document) highlight in black. KOReader draws our highlight in grey.

0.5
![14](https://user-images.githubusercontent.com/62179190/151036593-0d164d2e-19eb-4672-8749-73b463cee474.png)

1 (the same highlight!)

![15](https://user-images.githubusercontent.com/62179190/151036665-521a7373-db5c-49ad-a57f-da7d3b592cd0.png)

Reply:
Yes, I think it can be just 0.

Reply:
OK, so it's a delta y, 0 is the bottom of the line box, and 1 is the top of the line box.

> Yes, I think it can be just 0.

Was going to ask why you thought 0.1 was better than 0:

![image](https://user-images.githubusercontent.com/24273478/151037422-a9bb3916-6197-457f-a23e-0dc62a60931c.png)

I guess no black means they do superpose, which is what you wanted :) no ?

Reply:
(But may be with other PDF document with larger line heights, 0 would be the bottom of the taller line box ? No idea :/)
Anyway, let's go with 0 and see then.

Reply:
I think 0 is fine. With some larger interline PDF, with 0:
![image](https://user-images.githubusercontent.com/24273478/151039297-2f187f4e-64b0-478f-9297-0f4fd6c8b4f1.png)

0 is the bottom of the text box, and I guess in PDFs (unline crengine), we have text boxes absolutely positionned, and it's not the previous line box height that positions the next line box.